app/case-management/case-management.js

```diff
@@ -27,27 +30,93 @@ const caseDataDao = {
         try {
             let queryParts = {};
             queryParts = {
-                columns: "CONCAT(COALESCE(t4.given_name, ''), ' ', COALESCE(t4.middle_name,''), ' ', COALESCE(t4.family_name, '')) as person_name, t1.uuid as `user_uuid`, t5.value_reference as `location_uuid`",
-                table: "amrs.users",
+                columns: "CONCAT(COALESCE(t6.given_name, ''), ' ', COALESCE(t6.middle_name,''), ' ', COALESCE(t6.family_name, '')) as person_name, t1.uuid as `user_uuid`, t5.value_reference as `location_uuid`,count(t1.person_id) as `number_assigned`,t4.provider_id",
+                table: "amrs.person_attribute",
                 alias: 't1',
                 joins: [
-                    ['amrs.person', 't2', 't1.person_id = t2.person_id'],
-                    ['amrs.provider', 't3', 't2.person_id = t3.person_id'],
-                    ['amrs.person_name', 't4', 't2.person_id = t4.person_id'],
-                    ['amrs.provider_attribute', 't5', 't3.provider_id = t5.provider_id'],
+                    ['amrs.users', 't2', 't2.user_id = t1.value'],
+                    ['amrs.person', 't3', 't3.person_id = t2.person_id'],
+                    ['amrs.provider', 't4', 't4.person_id = t3.person_id'],
+                    ['amrs.provider_attribute', 't5', 't5.provider_id = t4.provider_id'],
+                    ['amrs.person_name', 't6', 't6.person_id = t3.person_id']
                 ],
-                where: ['t5.attribute_type_id = 1 and t5.value_reference = ?', 
-                params.locationUuid]
+                where: ['person_attribute_type_id=68 and t5.attribute_type_id = 1 and t5.value_reference = ?',
+                    params.locationUuid],
+                groupBy: 't4.provider_id'
             };
             return db.queryDb(queryParts);
         } catch (error) {
             return error;
         }
+    },
+    assignPatientsToCaseManagers: async (payload) => {
+        try {
+            let payloadData = getPayloadData(payload);
+            while (payloadData.length > 0) {
+                let data = payloadData.splice(0, 1);
+                let promisesArray = [];
+                data.forEach(d => {
+                    let payload = {
+                        "attributes": [
+                            {
+                                "attributeType": "9a6e12b5-98fe-467a-9541-dab11ad87e45",
+                                "value": d.user_uuid
+                            }
+                        ]
+                    }
+                    promisesArray.push(postPersonAttributes(payload, d.patient_uuid))
+                })
+                await Promise.all(promisesArray).then(function (r) {
```

Comment:
So we need to do 2 things here, 
1. Execute the promise serially so that we don't degrade openmrs
2. We need to report failed requests to the user. Promise.all will terminate if any of the requests fail.